catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveCommand.java

```diff
@@ -67,69 +73,74 @@ protected Object executeWithSubject() throws Exception {
     }
 
     private Object executeRemoveFromCache() throws Exception {
-        String[] idsArray = new String[ids.size()];
-        idsArray = ids.toArray(idsArray);
-        getCacheProxy().removeById(idsArray);
-
-        List idsList = Arrays.asList(ids);
-
-        printSuccessMessage(idsList + " successfully removed from cache.");
-
-        LOGGER.info(idsList + " removed from cache using catalog:remove command");
-
+        if (CollectionUtils.isNotEmpty(ids)) {
+            getCacheProxy().removeById(ids.toArray(new String[ids.size()]));
```

Comment:
✏️ Nit: `new String[0]` would actually be faster than `new String[id.size()]`

Reply:
Poor Josh. I was the one who change it from `new String[0]` to `new String[id.size()]`. I did not know the first was faster. Why is it so?

Reply:
Because when the array is `malloc`'d Java needs to null all the slots in it...and then the `toArray` call is just going to blow it away and make a new array anyway.